File: LongestWord/lw.py
# Longest Word solution in Python 2 for CodeEval.com by Steven A. Dunn

import sys

for line in open(sys.argv[1], 'r'):
  line = line.rstrip('\n').split(' ')
  counts = map(lambda s: int(len(s)), line)
  longest = line[counts.index(max(counts))]
  print longest

File: LongestWord/lw.rb
# longest word soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line = line.chomp.split(' ')
    max_length = -1
    max_word = ""
    line.each do |word|
        if word.length > max_length
            max_length = word.length
            max_word = word
        end
    end
    puts max_word
end